fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/NewNumberParmBehaviorUtilTest.java

```diff
@@ -0,0 +1,467 @@
+        FHIRRequestContext.get().setTenantId("default");
+    }
+
+    @Test
+    public void testPrecisionWithExact() throws FHIRPersistenceException {
+        // The spec states:
+        // When a comparison prefix in the set lgt, lt, ge, le, sa & eb is provided,
+        // the implicit precision of the number is ignored, and they are treated as
+        // if they have arbitrarily high precision
+        // However there is no 'lgt', means gt? (documented in https://jira.hl7.org/browse/FHIR-21216)
+
+        // gt - Greater Than
+        QueryParameter queryParm = generateParameter(SearchConstants.Prefix.GT, null, "1e3");
+        List<Object> expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        String expectedSql = "(Basic.NUMBER_VALUE > ? OR Basic.NUMBER_VALUE_HIGH > ?)";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+
+        // lt - Less Than
+        queryParm             = generateParameter(SearchConstants.Prefix.LT, null, "1e3");
+        expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedSql = "(Basic.NUMBER_VALUE < ? OR Basic.NUMBER_VALUE_LOW < ?)";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+
+        // ge - Greater than Equal
+        queryParm             = generateParameter(SearchConstants.Prefix.GE, null, "1e3");
+        expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedSql = "(Basic.NUMBER_VALUE >= ? OR Basic.NUMBER_VALUE_HIGH >= ?)";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+
+        // le - Less than Equal
+        queryParm             = generateParameter(SearchConstants.Prefix.LE, null, "1e3");
+        expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedSql = "(Basic.NUMBER_VALUE <= ? OR Basic.NUMBER_VALUE_LOW <= ?)";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+
+        // sa - starts after
+        queryParm             = generateParameter(SearchConstants.Prefix.SA, null, "1e3");
+        expectedBindVariables = new ArrayList<>();
+        expectedBindVariables.add(new BigDecimal("1E+3"));
+        // expectedBindVariables.add(new BigDecimal("1E+3"));
+        expectedSql = "Basic.NUMBER_VALUE_LOW > ?";
+        runTest(queryParm, expectedBindVariables, expectedSql);
+
+        // eb - Ends before
+        queryParm             = generateParameter(SearchConstants.Prefix.EB, null, "1e3");
+        expectedBindVariables = new ArrayList<>();
+        // expectedBindVariables.add(new BigDecimal("1E+3"));
```

Comment:
? what the comment for?

Reply:
As above. Trying to work out what the old code was doing.